documentation/architecture-decision-records/ADR0002-type_groups.md

```diff
@@ -0,0 +1,38 @@
+# Components to use utility function to get CSS for type groups
+## ADR-0002
+
+Status: Proposed
+* Deciders:
+  * Articles Team
+  * WS Team
+* Date: 2019-03-18
+
+Technical Story:
+
+World Service has a UX requirement that GEL Type Styles (Canon, Long Primer, Double Pica, etc) should allow for mutable values of font size and line height across services. The possible permutations of such values were grouped into eight different 'scripts': arabic, burmese, cjk_thai, hindi_nepali_bengali, latin, mundo, sinhala, and tamil, as shown in the tabs of [this spreadsheet](https://docs.google.com/spreadsheets/d/1WMY-rlSbekhB4pV3ojMuYoWDC71Qs93CB2wTfDwAsJg/edit#gid=1458344872)
+
+Additionally, a ninth script that contains the default values GEL for all type groups is also required.
+
+## Context and Problem Statement
+
+Currently, Psammead uses a hard-coded set of constants that express the Type Sizes for each Type Style as defined by GEL. We need a way to make these type sizes mutable and mappable to different services.
+
+## Decision Drivers
+
+* Type Styles (Canon, Trafalgar, etc) must allow for different values across services
+* Mappings for each style must be easily editable and updateable
+* A component should be able to fetch the required CSS for a Type Style by invoking its name (Canon, Trafalgar) and its current script (arabic, mundo, etc)
+
+## Considered Options
```

Comment:
Can you please talk about weight and tree shaking here

Reply:
Hey Drew. About tree shaking and weight, what exactly should be included? Is it how each package is handled based on the two?

Reply:
The way that styles have been exported, such that they can be passed into a function, allows for the rest of the 'scripts' to not be pulled into the application. We should address this, saying why we did this, and how it avoids pulling them all in

Reply:
Ok, this is better. Going to try and write this up.

Reply:
I've made some changes and they can be found [here](https://github.com/bbc/psammead/pull/364/commits/e76e60e9b848a3ed11536f6ddb4dbe0868f1c56a)